File: sdks/typescript/src/types/bot.ts
/**
 * Bot types
 */

import type { PaginationParams, Timestamps } from './common';

export interface Bot extends Timestamps {
  id: string;
  tenantId: string;
  name: string;
  description?: string;
  status: BotStatus;
  type: BotType;
  config: BotConfig;
  channelIds: string[];
  flowId?: string;
  agentId?: string;
  knowledgeBaseIds?: string[];
  metadata?: Record<string, unknown>;
}

export type BotStatus = 'active' | 'inactive' | 'draft';
export type BotType = 'flow' | 'ai' | 'hybrid';

export interface BotConfig {
  welcomeMessage?: string;
  fallbackMessage?: string;
  handoffMessage?: string;
  handoffTriggers?: string[];
  operatingHours?: OperatingHours;
  aiConfig?: AIBotConfig;
}

export interface OperatingHours {
  enabled: boolean;
  timezone: string;
  schedule: DaySchedule[];
  outsideHoursMessage?: string;
}

export interface DaySchedule {
  day: number; // 0-6 (Sunday-Saturday)
  enabled: boolean;
  startTime: string; // HH:mm
  endTime: string; // HH:mm
}

export interface AIBotConfig {
  model: string;
  temperature?: number;
  maxTokens?: number;
  systemPrompt?: string;
  useKnowledgeBase: boolean;
  enableStreaming?: boolean;
}

// Request types
export interface CreateBotInput {
  name: string;
  description?: string;
  type: BotType;
  config?: BotConfig;
  channelIds?: string[];
  flowId?: string;
  agentId?: string;
  knowledgeBaseIds?: string[];
  metadata?: Record<string, unknown>;
}

export interface UpdateBotInput {
  name?: string;
  description?: string;
  status?: BotStatus;
  config?: Partial<BotConfig>;
  channelIds?: string[];
  flowId?: string;
  agentId?: string;
  knowledgeBaseIds?: string[];
  metadata?: Record<string, unknown>;
}

export interface ListBotsParams extends PaginationParams {
  status?: BotStatus;
  type?: BotType;
  channelId?: string;
  search?: string;
}

File: sdks/go/types/bot.go
package types

// BotStatus type
type BotStatus string

const (
	BotStatusActive   BotStatus = "active"
	BotStatusInactive BotStatus = "inactive"
	BotStatusDraft    BotStatus = "draft"
)

// BotType type
type BotType string

const (
	BotTypeFlow   BotType = "flow"
	BotTypeAI     BotType = "ai"
	BotTypeHybrid BotType = "hybrid"
)

// Bot model
type Bot struct {
	ID               string                 `json:"id"`
	TenantID         string                 `json:"tenantId"`
	Name             string                 `json:"name"`
	Description      string                 `json:"description,omitempty"`
	Status           BotStatus              `json:"status"`
	Type             BotType                `json:"type"`
	Config           map[string]interface{} `json:"config"`
	ChannelIDs       []string               `json:"channelIds"`
	FlowID           string                 `json:"flowId,omitempty"`
	AgentID          string                 `json:"agentId,omitempty"`
	KnowledgeBaseIDs []string               `json:"knowledgeBaseIds,omitempty"`
	Metadata         map[string]interface{} `json:"metadata,omitempty"`
	Timestamps
}

// CreateBotInput for creating bots
type CreateBotInput struct {
	Name             string                 `json:"name"`
	Description      string                 `json:"description,omitempty"`
	Type             BotType                `json:"type"`
	Config           map[string]interface{} `json:"config,omitempty"`
	ChannelIDs       []string               `json:"channelIds,omitempty"`
	FlowID           string                 `json:"flowId,omitempty"`
	AgentID          string                 `json:"agentId,omitempty"`
	KnowledgeBaseIDs []string               `json:"knowledgeBaseIds,omitempty"`
	Metadata         map[string]interface{} `json:"metadata,omitempty"`
}

// UpdateBotInput for updating bots
type UpdateBotInput struct {
	Name             string                 `json:"name,omitempty"`
	Description      string                 `json:"description,omitempty"`
	Status           BotStatus              `json:"status,omitempty"`
	Config           map[string]interface{} `json:"config,omitempty"`
	ChannelIDs       []string               `json:"channelIds,omitempty"`
	FlowID           string                 `json:"flowId,omitempty"`
	AgentID          string                 `json:"agentId,omitempty"`
	KnowledgeBaseIDs []string               `json:"knowledgeBaseIds,omitempty"`
	Metadata         map[string]interface{} `json:"metadata,omitempty"`
}

// ListBotsParams for list request
type ListBotsParams struct {
	PaginationParams
	Status    BotStatus `json:"status,omitempty"`
	Type      BotType   `json:"type,omitempty"`
	ChannelID string    `json:"channelId,omitempty"`
	Search    string    `json:"search,omitempty"`
}
